workers/unity/Packages/io.improbable.worker.sdk/CEventTrace.cs

```diff
@@ -18,13 +18,23 @@ internal unsafe class CEventTrace
          * Data for an event. This is a collection of key-value pairs (fields). Use EventData* functions to
          * read or write fields.
          */
-        public class EventData : CptrHandle
+        public class EventDataHandle : CptrHandle
         {
             protected override bool ReleaseHandle()
             {
                 EventDataDestroy(handle);
                 return true;
             }
+
+            internal void SetUnderlying(IntPtr newHandle)
+            {
+                SetHandle(newHandle);
```

Comment:
This allows you to override the previous handle, this could lead to a leak?

Reply:
Yes, this could definitely lead to a leak, however, it's only used in one place:
https://github.com/spatialos/gdk-for-unity/pull/1452/files#diff-f0724dab9c5e9627ca1d44f44414bd32R120

It was the only way I could figure out to convert the EventData from the internal to external representations. In the `TraceEventData` internal constructor, we take a copy of the old handle and call `EventDataDestroy` to make sure any allocated memory is released

Reply:
What about making the function private, and add an additional constructor?

Reply:
Yep, that seems to work, it made it a bit more clear I think 👍